nautobot/ipam/tests/test_api.py

```diff
@@ -1006,20 +1015,109 @@ def test_list_available_vlans(self):
-        VLAN.objects.create(name="vlan_17", vid=17, status=Status.objects.first(), vlan_group=vg)
-        unused_vids = [6, 7, 8, 9, 15, 16, 18, 19, 20]
-
-        url = reverse("ipam-api:vlangroup-available-vlans", kwargs={"pk": vg.pk})
+        url = reverse("ipam-api:vlangroup-available-vlans", kwargs={"pk": self.vlan_group.pk})
         self.add_permissions("ipam.view_vlangroup")
 
         # Retrieve all available VLAN IDs
         response = self.client.get(url, **self.header)
 
-        self.assertEqual(response.data["results"], unused_vids)
-        self.assertEqual(response.data["count"], len(unused_vids))
+        self.assertEqual(response.data["results"], self.unused_vids)
+        self.assertEqual(response.data["count"], len(self.unused_vids))
+
+    def test_create_single_available_vlan(self):
+        """
+        Test creation of the first available VLAN within a VLANGroup.
+        """
+        cf = CustomField.objects.create(key="sor", label="Source of Record Field", type="text")
+        cf.content_types.add(ContentType.objects.get_for_model(VLAN))
+        url = reverse("ipam-api:vlangroup-available-vlans", kwargs={"pk": self.vlan_group.pk})
+        self.add_permissions(
+            "ipam.view_vlangroup",
+            "ipam.change_vlangroup",
+            "ipam.add_vlangroup",
+            "ipam.change_vlan",
+            "ipam.view_vlan",
+            "ipam.add_vlan",
+            "extras.view_status",
+        )
+
+        # Create all nine available VLANs with individual requests
+        for unused_vid in self.unused_vids:
+            data = {
+                "name": f"VLAN_{unused_vid}",
+                "description": f"Test VLAN {unused_vid}",
+                "status": self.default_status.pk,
+                "custom_fields": {"sor": "Nautobot"},
+            }
+            response = self.client.post(url, data, format="json", **self.header)
+            self.assertHttpStatus(response, status.HTTP_201_CREATED)
+            self.assertEqual(response.data["results"][0]["name"], data["name"])
+            self.assertEqual(response.data["results"][0]["vid"], unused_vid)
+            self.assertEqual(response.data["results"][0]["description"], data["description"])
+            self.assertIn("custom_fields", response.data["results"][0])
+            self.assertIn("sor", response.data["results"][0]["custom_fields"])
+            self.assertEqual("Nautobot", response.data["results"][0]["custom_fields"]["sor"])
+
+        # Try to create one more VLAN
+        response = self.client.post(url, {"name": "UTILIZED_VLAN_GROUP"}, format="json", **self.header)
+        self.assertHttpStatus(response, status.HTTP_204_NO_CONTENT)
+        self.assertIn("detail", response.data)
+        self.assertIn(
+            f"An insufficient number of VLANs are available within the VLANGroup {self.vlan_group}",
+            response.data["detail"],
+        )
+
+    def test_create_multiple_available_vlans(self):
```

Comment:
Might be good to add a couple more tests:

- POST to available-vlans and explicitly specifying the vid (as in the docs)
  - POST to available-vlans with multiple entries trying to explicitly specify the same vid (should be rejected)
- POST to available-vlans and explicitly specifying the vlan_group (should probably be rejected?)

Reply:
Done 👍